src/VisualStudio/Core/Def/LanguageServer/Handler/CustomMessage/CustomMessage.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+using Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CustomMessage;
+
+internal class CustomMessage(JsonNode message, TextDocumentIdentifier textDocument, Position[] positions)
```

Comment:
1. seal
2. docs.
3. what is the Position[] for?  Unclear what the purpose of this is
4. similar question about the textDocument.  there are lots of messages not referring to a particular doc, so it seems odd to bake this in.